src/security-council-mgmt/SecurityCouncilNominationsGovernor.sol

```diff
@@ -0,0 +1,225 @@
+        revert("TODO");
+    }
+
+    // unsupported because it is supposed to return for, against, abstain. but that doesn't make sense for nominations
+    function proposalVotes(
+        uint256
+    ) public pure override(GovernorCountingSimpleUpgradeable) returns (uint256, uint256, uint256) {
+        revert("Unsupported");
+    }
+
+    // there is no minimum quorum for nominations, so we just return true
+    function _quorumReached(uint256) internal pure override(GovernorCountingSimpleUpgradeable, GovernorUpgradeable) returns (bool) {
+        return true;
+    }
+
+    // the vote always succeeds, so we just return true
+    function _voteSucceeded(uint256) internal pure override(GovernorCountingSimpleUpgradeable, GovernorUpgradeable) returns (bool) {
+        return true;
+    }
+
+    function _countVote(
+        uint256 proposalId,
+        address account,
+        uint8,
+        uint256 weight,
+        bytes memory params
+    ) internal virtual override(GovernorCountingSimpleUpgradeable, GovernorUpgradeable) {
+        // let's say params is (address candidate, uint256 tokens)
+        (address candidate, uint256 tokens) = abi.decode(params, (address, uint256));
+
+        require(_isCandidateEligible(proposalId, candidate), "Candidate is not eligible");
+
+        NominationElection storage election = nominationElections[proposalId];
+
+        // weight is the number of tokens that account has at the time of the vote
+        // make sure tokens + previously used tokens is less than or equal to weight
+        uint256 previouslyUsedTokens = election.nominationsTokensUsed[account];
+        require(tokens + previouslyUsedTokens <= weight, "Not enough tokens to cast this vote");
+
+        // add to nominationsTokensUsed
+        election.nominationsTokensUsed[account] = previouslyUsedTokens + tokens;
+
+        // add tokens to the candidate
+        uint256 oldVotesForCandidate = election.nominationsVotes[candidate];
+        election.nominationsVotes[candidate] = oldVotesForCandidate + tokens;
+
+        // if this vote put the candidate over the line, push to successfullyNominatedCandidates
+        uint256 votesNeeded = quorum(proposalSnapshot(proposalId));
+        if (oldVotesForCandidate < votesNeeded && oldVotesForCandidate + tokens >= votesNeeded) {
+            election.successfullyNominatedCandidates.push(candidate);
+            // emit some event like CandidateSuccessfullyNominated(proposalId, candidate);
+        }
+    }
+
+    ////////////// End GovernorCountingSimple Overrides //////////////
+
+    ////////////// GovernorTimelockControl Overrides //////////////
+
+    // should these view functions just revert?
+
+    function state(uint256 proposalId) public view virtual override returns (ProposalState) {
+        return GovernorUpgradeable.state(proposalId);
+    }
+
+    function timelock() public view virtual override returns (address) {
```

Comment:
I think GovernorTimelockControlUpgradeable assumes/requires a timelock here (which we may want anyway, to add in the compliance period)

Reply:
...either way, we'll want functionality for the foundation (not the council!) to reject non-compliant nominees (as per the constitution)

Reply:
potential implementation of this feature in 4d5dc07b6e6bb706feb3b372a5de042d20422598